src/Workspaces/Core/Portable/Utilities/VoidResult.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Roslyn.Utilities
+{
+    /// <summary>
+    /// Explicitly indicates result is void
+    /// </summary>
+    internal class VoidResult
```

Comment:
:bulb: This can be simplified to something resembling this:

http://referencesource.microsoft.com/#System.Core/System/Threading/Tasks/TaskExtensions.cs,217